### PR TITLE
Framework: Drop deprecations slated for 3.1 removal

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to remove deprecations slated for removal in the upcoming 3.1.0 release.

Specifically, this includes:

 - All components in `wp.blocks.*` are removed. Please use `wp.editor.*` instead.
 - `wp.blocks.withEditorSettings` is removed. Please use the data module to access the editor settings `wp.data.select( "core/editor" ).getEditorSettings()`.
 - All DOM utils in `wp.utils.*` are removed. Please use `wp.dom.*` instead.
 - `isPrivate: true` has been removed from the Block API. Please use `supports.inserter: false` instead.
 - `wp.utils.isExtraSmall` function removed. Please use `wp.viewport.isExtraSmall` instead.
 - `getEditedPostExcerpt` selector removed (`core/editor`). Use `getEditedPostAttribute( 'excerpt' )` instead.

See: https://wordpress.org/gutenberg/handbook/reference/deprecated/#3-1-0

__Testing instructions:__

Verify that there are no regressions in impacted behavior, and importantly that no references to deprecated behaviors exist in core code.

Notably, test the excerpt field, which had not been previously upgraded until this pull request.

### CODE REVIEW

File: edit-post/index.js [hunk 1]
```diff
@@ -50,15 +50,6 @@ export function reinitializeEditor( postType, postId, target, settings, override
  * @return {Object} Editor interface.
  */
 export function initializeEditor( id, postType, postId, settings, overridePost ) {
-	if ( 'production' !== process.env.NODE_ENV ) {
-		// Remove with 3.0 release.
```
Comment (gziolo) [inline edit-post/index.js]: Yeah, thanks 👍

**Review by gziolo — COMMENTED** [review]
I believe there were also PHP changes introduced to make the following polyfill work:

> All components in `wp.blocks.*` are removed. Please use `wp.editor.*` instead.

**Comment by gziolo** [comment]
I removed PHP part with 8bf8d21. @youknowriad can you double check if we didn't miss anything?

**Review by gziolo — APPROVED** [review]
LGTM, I tested the editor with the changes applied and it still works 👍

**Comment by gziolo** [comment]
One more commit added 818acd1. Removed deprecated utils :)
Now, it should be good for real.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

File: utils/deprecated.js [hunk 2]
```diff
@@ -57,14 +30,3 @@ export function deprecated( ...params ) {
 
 	return originalDeprecated( ...params );
 }
-
-// viewport
-export function isExtraSmall() {
-	originalDeprecated( 'wp.utils.isExtraSmall', {
```
Comment (aduth) [inline utils/deprecated.js]: Unfortunately the custom ESLint rule is not smart enough to capture anything except the default function name `deprecated` 😅

Comment (gziolo) [reply utils/deprecated.js]: Right, I was to creative renaming its name :)

**Pull request merged**